diffsims/__init__.py

```diff
@@ -22,12 +22,16 @@
 
 import numpy as np
 
-from .generators.diffraction_generator import DiffractionGenerator
+from .generators.diffraction_generator import DiffractionGenerator, AtomicDiffractionGenerator
```

Comment:
I think we should be able to come up with better names than this - I think the difference is really that we have a structure factor based simulation and an FFT based simulation, we also have (a bit hidden in the code) a real space sum of scattering from many atoms solution. So I think we should either have one DiffractionGenerator with multiple methods or name more based around the simulation approach. I prefer the latter here so maybe we could go with:

SFDiffractionGenerator
FFTDiffractionGenerator
RSDifractionGenerator

Or we could change and have:

StructureFactorSimulator
FFTSimulator
RealSpaceSimulator

What do others think?

Reply:
I agree with the latter. The main problem I see with trying to merge them into one object is that, as far as I understand it, they just take different inputs (whole crystal vs unit cell) and give different outputs (single diffraction pattern vs complete list of Bragg angles).

I'm not entirely sure what your 'real space' simulator is but I'm not sure about using FFT as part of the naming system. If you add a multi-slice implementation later then I think that will also use FFTs but I guess you'd also want to give it another name? I think these are all fundamentally different models and a choice of backend should be a different parameter/method.

I'm not sure what would be a better name, I guess you will both know much better than me. FirstBorn, Ewald, or something with Kinematic?.. Other than that, I don't have much preference over the general naming format.